dds/idl/marshal_generator.cpp

```diff
@@ -769,801 +1106,254 @@ namespace {
             << "    if (!(strm >> fa)) {\n";
         } else if (elem_cls & CL_STRING) {
           if (elem_cls & CL_BOUNDED) {
-            const string args = string("seq[i]") + (use_cxx11 ? ", " : ".out(), ") + bounded_arg(elem);
-            be_global->impl_ << "    if (!(strm " << ">> " << getWrapper(args, elem, WD_INPUT) << ")) {\n";
+            const string args = elem_access + (use_cxx11 ? ", " : ".out(), ") + bounded_arg(elem);
+            be_global->impl_ <<
+              "    if (!(strm >> " << getWrapper(args, elem, WD_INPUT) << ")) {\n";
           } else {
             const string getbuffer =
               (be_global->language_mapping() == BE_GlobalData::LANGMAP_NONE)
               ? ".get_buffer()" : "";
-            be_global->impl_ << "if (!(strm >> seq" + getbuffer + "[i])) {\n";
+            be_global->impl_ <<
+              "    if (!(strm >> " << value_access << getbuffer << "[i])) {\n";
           }
-        } else if (use_cxx11 && (elem_cls & (CL_ARRAY | CL_SEQUENCE))) {
-          be_global->impl_ <<
-            "      if (!(strm >> IDL::DistinctType<" << cxx_elem << ", "  <<
-            elem_underscores << "_tag>(seq[i])" << ")) {\n";
         } else {
-          be_global->impl_ << "   if (!(strm >> seq[i])) {\n";
+          Wrapper elem_wrapper(elem, cxx_elem, value_access + "[i]", false);
+          elem_wrapper.nested_key_only_ = nested_key_only;
+          Intro intro;
+          elem_wrapper.done(&intro);
+          const std::string indent = "    ";
+          intro.join(be_global->impl_, indent);
+          be_global->impl_ <<
+            indent << " if (!(strm >> " << elem_wrapper.ref() << ")) {\n";
         }
+
         const std::string seq_resize_func = use_cxx11 ? "resize" : "length";
 
         if (try_construct == tryconstructfailaction_use_default) {
           be_global->impl_ <<
-            "     " << type_to_default(elem, "seq[i]") <<
+            "     " << type_to_default(elem, elem_access) <<
             "           strm.set_construction_status(Serializer::ConstructionSuccessful);\n";
         } else if ((try_construct == tryconstructfailaction_trim) && (elem_cls & CL_BOUNDED) &&
                    (elem_cls & (CL_STRING | CL_SEQUENCE))) {
           if (elem_cls & CL_STRING){
-            const string check_not_empty = use_cxx11 ? "!seq[i].empty()" : "seq[i].in()";
-            const string get_length = use_cxx11 ? "seq[i].length()" : "ACE_OS::strlen(seq[i].in())";
-            const string inout = use_cxx11 ? "" : ".inout()";
+            const std::string check_not_empty =
+              use_cxx11 ? "!" + elem_access + ".empty()" : elem_access + ".in()";
+            const std::string get_length =
+              use_cxx11 ? elem_access + ".length()" : "ACE_OS::strlen(" + elem_access + ".in())";
+            string inout = use_cxx11 ? "" : ".inout()";
```

Comment:
Why not const?

Reply:
Unless I messed up merging (which is entirely possible), I think this was how the code was originally.

Reply:
It was `const` before.  Also this whole file can leave off the `std::` qualifier on `string`.